### PR TITLE
Archive client staging

### PR DESCRIPTION
**Pull request by unparalleled-js**
### Description of Change ###

I've updated ArchiveClient and ArchiveModule with functions to obtain archive information from query values accepted by the Archive API (https://www.crashplan.com/apidocviewer/#Archive)

User-facing functions are;
- `sdk.archive.get_by_archive_guid`
- `sdk.archive.get_by_device_guid`
- `sdk.archive.get_by_user_uid`
- `sdk.archive.get_by_store_point_id`
- `sdk.archive.get_by_destination_guid`
- `sdk.archive.get_by_plan_uid`
- `sdk.archive.get_by_user_uid_list`

For updates to ArchiveClient, there were some considerations;

- Can't specify an archiveGuid value through parameters
- Query values are numbers except the list of userUid values, which are strings.

Because of the above, the following methods have been added;
- `archive_client.get_single_archive` 
  - Queries a single archiveGuid
- `archive_client.get_archives_from_value`
  - Handles the vast majority of query values, and sets params dynamically from `id_type`.
- `archive_client.get_archives_from_user_uid_list`
  - Handles the comma-separated list of userUid strings

### Testing Procedure ###

Test have been written against the 3 new methods in ArchiveClient. Each test provide it's own parameters.

### CODE REVIEW

**Comment by unparalleled-js** [comment]
I didn't do 99% of this work, so I approve this PR (but can't actually approve it)

**Comment by alanag13** [comment]
@DiscoRiver would you mind changing your tests to all use `assert_called_once_with` (vs `assert_called_with`)?
Otherwise this looks good to me.

**Comment by unparalleled-js** [comment]
LGTM

File: CHANGELOG.md [hunk 1]
```diff
@@ -23,9 +23,14 @@ how a consumer would use the library (e.g. adding unit tests, updating documenta
 
 ### Added
 
-- [.netrc](https://ec.haxx.se/usingcurl/usingcurl-netrc) support. Calling `py42.sdk.from_local_account()` with no username and password will now result in py42 attempting to authenticate via an entry in your `.netrc` file, if you have one.
+- Methods for obtaining archive information:
+    - `sdk.archive.get_by_archive_guid`
```
Comment (unparalleled-js) [inline CHANGELOG.md:27]: Some of these should have the workd `all` in there, like `get_all_by_device_guid`

Comment (unparalleled-js) [reply CHANGELOG.md:27]: get_all_by_user_uid

Comment (unparalleled-js) [reply CHANGELOG.md:27]: get_all_by_dest..

**Pull request merged**